lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/Modal.java

```diff
@@ -19,24 +23,44 @@
     private ModalListener modalListener;
     @Nullable private Promise dismissPromise;
 
+    private ModalAnimator animator;
+
     public Modal(final ViewController viewController, ModalListener modalListener) {
         this.viewController = viewController;
         dialog = new Dialog(viewController.getActivity(), R.style.Modal);
         this.modalListener = modalListener;
         dialog.setOnKeyListener(this);
         dialog.setOnDismissListener(this);
         dialog.setOnShowListener(this);
+        animator = new ModalAnimator(viewController.getActivity());
+        animator.setOptions(viewController.options.animationsOptions);
```

Comment:
Options should be passed in constructor as they are used to configure the animator.